### PR TITLE
feature: Wireless Sensors Overhaul 

### PR DESCRIPTION
**Pull request by murrant**
Includes client counts for ios and unifi
Graphing could use some improvement.
Alerting and threshold ui not implemented

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

TODO:
- [x] Discovery
- [x] Polling
- [x] UI (Graphs and Custom thresholds)
- [x] Alerting
- [x] Documentation
- [x] Testing - Deferred too much is needed still

### CODE REVIEW

**Comment by laf** [comment]
I've not checked through the code much but 2 things:

1. I think we should do all 'vendor' work in one file (we should do this in sensors) so includes/discovery/wireless/unifi.inc.php would contain all wireless sensor work rather than splitting it out (imho).

2. The code that's commented out for alerting on eventlog, in the sensors code that's a hang up from the old alerting system, should we use it still as rules are defined by the user so they may not care about this info, if they do, they create an alert rule?

**Comment by laf** [comment]
Would this fit into this work: https://community.librenms.org/t/store-ubnt-airfiber-capacity-data-in-the-database/1046

p.s Awesome work :)

**Comment by murrant** [comment]
Indeed, these will all be alertable.  With default rules and user overrides hopefully.

**Comment by laf** [comment]
If anyone has an snmpwalk of the data this would use so I can test that would be ace.

**Comment by murrant** [comment]
The Cisco walk I have is from Rosiak it is not annonymized.  Unified is from a live device.

File: sql-schema/187.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+CREATE TABLE `wireless_sensors` (sensor_id INT(11) PRIMARY KEY NOT NULL AUTO_INCREMENT, sensor_deleted TINYINT(1) DEFAULT '0' NOT NULL, sensor_class VARCHAR(64) NOT NULL, device_id INT(11) unsigned DEFAULT '0' NOT NULL, sensor_index VARCHAR(64), sensor_type VARCHAR(255) NOT NULL, sensor_descr VARCHAR(255), sensor_divisor INT(11) DEFAULT '1' NOT NULL, sensor_multiplier INT(11) DEFAULT '1' NOT NULL, sensor_current FLOAT, sensor_limit FLOAT, sensor_limit_warn FLOAT, sensor_limit_low FLOAT, sensor_limit_low_warn FLOAT, sensor_alert TINYINT(1) DEFAULT '1' NOT NULL, sensor_custom ENUM('No', 'Yes') DEFAULT 'No' NOT NULL, entPhysicalIndex VARCHAR(16), lastupdate TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL, sensor_prev FLOAT, sensor_oids VARCHAR(255) NOT NULL, sensor_aggregator VARCHAR(16) DEFAULT 'sum' NOT NULL, access_point_id INT(11), entPhysicalIndex_measured VARCHAR(16), CONSTRAINT wireless_sensors_device_id_foreign FOREIGN KEY (device_id) REFERENCES `devices` (device_id) ON DELETE CASCADE);
+CREATE INDEX sensor_class ON `wireless_sensors` (sensor_class);
+CREATE INDEX sensor_host ON `wireless_sensors` (device_id);
+CREATE INDEX sensor_type ON `wireless_sensors` (sensor_type)
```
Comment (laf) [inline sql-schema/187.sql]: Needs ; at the end.

**Comment by murrant** [comment]
I'm reorganizing the code structure of this, so hold tight.

**Comment by murrant** [comment]
Updated!

Left to implement:
UI (Graphs and Custom thresholds)
Alerting
Testing

**Comment by FTBZ** [comment]
This is a really good news and I'm available to help and test if needed (Cisco WLC here). Can this refactor used to add users by SSID like Cacti does?

**Comment by murrant** [comment]
FTBZ: Mikrotik displays user counts by SSID.  Unifi, I'm considering adding both radio and ssid counts for clients.  Basically, we can break them out as much as the snmp data allows :)

**Comment by laf** [comment]
@murrant Feel free to merge this when you're happy.

**Comment by murrant** [comment]
After Sunday release sometime.

**Comment by laf** [comment]
#6327 possibly has data you can use.

**Comment by murrant** [comment]
I'm almost ready to push this, but I would like to get the power/signal issue figured out.
https://community.librenms.org/t/improve-wifi-collection-and-graphing-code/453/13?u=murrant

**Comment by murrant** [comment]
Preview :)

image

**Comment by murrant** [comment]
Thanks to @FTBZ, @Rosiak, and robdb on IRC for test data!

**Comment by laf** [comment]
image

**Pull request merged**